images/access-setup/content/bin/setup_compute.sh

```diff
@@ -107,6 +107,14 @@ init() {
   mkdir -p "$credentials_dir"
 }
 
+check_prerequisites() {
+  # Check that argocd has been installed
+  if ! kubectl get applications.argoproj.io --ignore-not-found; then
+    echo "ArgoCD must be deployed on the cluster first" >&2
+    exit 1
+  fi
+}
+
```

Comment:
Having the API does not guarantee that the operator is installed on the cluster. OLM for instance would not remove the CRD when the operator is uninstalled by default.
If I was to check the API I would rather call `kubectl api-resources` as it is to check the APIs available rather than making a get and waiting it to fail.
[Here](https://github.com/openshift-pipelines/pipelines-service/blob/main/images/cluster-setup/install.sh#L125-L147) for instance the availability of the deployment is checked.

Reply:
I had an issue with `grep -q` not behaving as expected. I used `grep -c` instead.